lib/parserm/parserM.ml

```diff
@@ -0,0 +1,27 @@
+module type Distinguishable = sig
+  type t
+  val equal : t -> t -> bool
+end
+
+module type ParserMSig = sig
+  type token
+  type 'a parser_m = token list -> ('a * token list, string) result 
+  val (>>=) : 'a parser_m -> ('a -> 'b parser_m) -> 'b parser_m
+  val (let*) : 'a parser_m -> ('a -> 'b parser_m) -> 'b parser_m
```

Comment:
https://github.com/janestreet/ppx_let

`and` is also better than `in`: the latter expresses a sequential dependency which need not exist in some implementations, even if this implementation is purely sequential